include/exadg/grid/grid_utilities.h

```diff
@@ -290,118 +297,129 @@ create_coarse_triangulations(
+          fine_triangulation,
+          BalancedGranularityPartitionPolicy<dim>(
+            dealii::Utilities::MPI::n_mpi_processes(fine_triangulation.get_communicator())));
+    }
+    else
+    {
+      AssertThrow(false, dealii::ExcMessage("Invalid parameter triangulation_type."));
+    }
 
-    coarse_triangulations =
-      dealii::MGTransferGlobalCoarseningTools::create_geometric_coarsening_sequence(
-        fine_triangulation,
-        BalancedGranularityPartitionPolicy<dim>(
-          dealii::Utilities::MPI::n_mpi_processes(fine_triangulation.get_communicator())));
+    // deal.II adds the fine triangulation as the last entry of the vector. According to our
+    // convention in ExaDG, we only include the triangulations coarser than the fine level. Hence,
+    // we remove the last entry of the vector.
+    coarse_triangulations.pop_back();
 
     coarse_periodic_face_pairs.resize(coarse_triangulations.size());
     for(unsigned int level = 0; level < coarse_periodic_face_pairs.size(); ++level)
       coarse_periodic_face_pairs[level] = fine_periodic_face_pairs;
   }
   else if(data.triangulation_type == TriangulationType::FullyDistributed)
   {
-    // resize the empty coarse_triangulations and coarse_periodic_face_pairs vectors
-    coarse_triangulations = std::vector<std::shared_ptr<dealii::Triangulation<dim> const>>(
-      fine_triangulation.n_global_levels());
-
-    coarse_periodic_face_pairs =
-      std::vector<PeriodicFacePairs<dim>>(fine_triangulation.n_global_levels());
-
-    // lambda function for creating the coarse triangulations
-    auto const lambda_create_level_triangulation =
-      [&](PeriodicFacePairs<dim> &  level_periodic_face_pairs,
-          unsigned int              refine_global,
-          std::vector<unsigned int> refine_local) {
-        auto level_triangulation =
-          std::make_shared<dealii::parallel::fullydistributed::Triangulation<dim>>(
-            fine_triangulation.get_communicator());
-
-        GridUtilities::create_triangulation<dim>(*level_triangulation,
-                                                 level_periodic_face_pairs,
-                                                 data,
-                                                 lambda_create_triangulation,
-                                                 refine_global,
-                                                 refine_local);
-
-        return level_triangulation;
-      };
-
-    // we start with the finest level
-    unsigned int              level        = fine_triangulation.n_global_levels() - 1;
-    std::vector<unsigned int> refine_local = vector_local_refinements;
-
-    // make the last entry of the coarse_triangulations point to the fine_triangulation
-    coarse_triangulations[level].reset(&fine_triangulation, [](auto *) {
-      // empty deleter, since fine_triangulation is an external field
-      // and its destructor is called somewhere else
-    });
```

Comment:
I think we no longer need this "hard to understand" code (also not somewhere else). In my opinion this also shows that appending the vector of coarse triangulations by the fine triangulation was somewhat artificial in the previous code version.